cmd/ocp-metadata/cmd/root.go

```diff
@@ -0,0 +1,171 @@
+including cluster information, node details, networking configuration, and more.
+
+The tool supports multiple output formats (json, yaml, table) and retrieves:
+- Cluster name, platform, and type
+- OCP and Kubernetes versions
+- Node information (masters, workers, infra)
+- Network configuration (SDN type, IPSec)
+- Architecture information
+- FIPS and publish configuration`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		// Initialize the metadata client
+		restConfig, err := getRestConfig()
+		if err != nil {
+			return fmt.Errorf("failed to get Kubernetes config: %w", err)
+		}
+
+		metadata, err := ocpmetadata.NewMetadata(restConfig)
+		if err != nil {
+			return fmt.Errorf("failed to initialize metadata client: %w", err)
+		}
+
+		// Get cluster metadata
+		clusterMetadata, err := metadata.GetClusterMetadata()
+		if err != nil {
+			return fmt.Errorf("failed to get cluster metadata: %w", err)
+		}
+
+		return outputData(clusterMetadata)
+	},
+}
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+func Execute() error {
+	return rootCmd.Execute()
+}
+
+func init() {
+	// Global flags
+	rootCmd.Flags().StringVar(&kubeconfig, "kubeconfig", "", "Path to kubeconfig file (default is $HOME/.kube/config)")
+	rootCmd.Flags().StringVarP(&outputFormat, "output", "o", "table", "Output format (json|yaml|table)")
+}
+
+// getRestConfig returns a Kubernetes REST config
+func getRestConfig() (*rest.Config, error) {
+	if kubeconfig == "" {
+		if home := homedir.HomeDir(); home != "" {
+			kubeconfig = filepath.Join(home, ".kube", "config")
+		}
+	}
+
+	// Try in-cluster config first
+	if config, err := rest.InClusterConfig(); err == nil {
+		return config, nil
+	}
+
+	// Fall back to kubeconfig file
+	return clientcmd.BuildConfigFromFlags("", kubeconfig)
+}
+
+// outputData formats and prints data based on the output format
+func outputData(data interface{}) error {
+	switch outputFormat {
+	case "json":
+		encoder := json.NewEncoder(os.Stdout)
+		encoder.SetIndent("", "  ")
+		return encoder.Encode(data)
+	case "yaml":
+		encoder := yaml.NewEncoder(os.Stdout)
+		defer encoder.Close()
+		return encoder.Encode(data)
+	case "table":
+		return outputTable(data)
+	default:
+		return fmt.Errorf("unsupported output format: %s", outputFormat)
+	}
+}
+
+// outputTable handles table format output for different data types
+func outputTable(data interface{}) error {
+	switch v := data.(type) {
+	case ocpmetadata.ClusterMetadata:
+		return printClusterMetadataTable(v)
+	case map[string]interface{}:
+		return printMapTable(v)
+	case string:
+		fmt.Println(v)
+		return nil
+	default:
+		// Fall back to JSON for complex types
+		encoder := json.NewEncoder(os.Stdout)
+		encoder.SetIndent("", "  ")
+		return encoder.Encode(data)
+	}
+}
+
+// printClusterMetadataTable prints cluster metadata in table format
+func printClusterMetadataTable(metadata ocpmetadata.ClusterMetadata) error {
```

Comment:
Not sure how much weight-age we have for tabular data, format specifiers might make things look ugly. suggesting the same [tablewriter](https://github.com/vishnuchalla/k8s-netperf/blob/main/pkg/results/result.go#L17) as we have in k8-netperf.